NewDawn/NewDawn/MainPageViewController.swift

```diff
@@ -16,12 +16,16 @@ class MainPageViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        if ProfileIndexUtil.loadProfileIndex() >= USER_DUMMY_DATA.count {
+        if TimerUtil.isOutdated(){
+            UserProfileBuilder.fetchAndStoreUserProfiles()
```

Comment:
Does this solve the problem?

Reply:
I realize that my getUserProfileListFromLocalStorage does not need that process session logic.
But that semaphore technique is really awesome. Everything works fine as of this commit.

Reply:
> Better to use LocalStorageUtil function to do this instead of encoding and set them manually.
However, I met some problems when I try to resolve this...

Reply:
This PR is getting me crazy.

Reply:
What's the other problems?

Reply:
It is resolved.